gratipay/models/participant.py

```diff
@@ -416,11 +417,11 @@ def add_email(self, email, resend_threshold='3 minutes'):
             nonce = self.db.one("""
                 UPDATE emails
                    SET verification_start=%s
-                 WHERE participant=%s
+                 WHERE participant_id=%s
                    AND address=%s
                    AND verified IS NULL
              RETURNING nonce
-            """, (verification_start, self.username, email))
+            """, (verification_start, self.id, email))
             if not nonce:
                 return self.add_email(email)
 
```

Comment:
Haha. This is convincing enough. Most email test are failing. Below is one,

```
====================================================================== FAILURES ======================================================================
__________________________________________________ TestClosing.test_cpi_clears_personal_information __________________________________________________

args = (<test_close.TestClosing testMethod=test_cpi_clears_personal_information>, <MagicMock name=u'_mailer' id='46937897223696'>), keywargs = {}
extra_args = [<MagicMock name=u'_mailer' id='46937897223696'>], entered_patchers = [<mock._patch object at 0x2ab0944ed990>]
exc_info = (<type 'exceptions.RuntimeError'>, RuntimeError('maximum recursion depth exceeded while encoding a JSON object',), <traceback object at 0x2ab0949db950>)
patching = <mock._patch object at 0x2ab0944ed990>, arg = <MagicMock name=u'_mailer' id='46937897223696'>

    @wraps(func)
    def patched(*args, **keywargs):
        # don't use a with here (backwards compatability with Python 2.4)
        extra_args = []
        entered_patchers = []
    
        # can't use try...except...finally because of Python 2.4
        # compatibility
        exc_info = tuple()
        try:
            try:
                for patching in patched.patchings:
                    arg = patching.__enter__()
                    entered_patchers.append(patching)
                    if patching.attribute_name is not None:
                        keywargs.update(arg)
                    elif patching.new is DEFAULT:
                        extra_args.append(arg)
    
                args += tuple(extra_args)
>               return func(*args, **keywargs)

env/lib/python2.7/site-packages/mock.py:1201: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/py/test_close.py:147: in test_cpi_clears_personal_information
    alice.add_email('alice@example.net')
gratipay/models/participant.py:426: in add_email
    return self.add_email(email)
gratipay/models/participant.py:426: in add_email
    return self.add_email(email)
gratipay/models/participant.py:426: in add_email
    return self.add_email(email)
gratipay/models/participant.py:426: in add_email
    return self.add_email(email)
gratipay/models/participant.py:426: in add_email
    return self.add_email(email)
gratipay/models/participant.py:426: in add_email
    return self.add_email(email)
gratipay/models/participant.py:426: in add_email
    return self.add_email(email)

```